Applications/Tizen_native/CustomShortcut/src/data.c

```diff
@@ -237,9 +246,59 @@ int data_extract_feature(appdata_s *ad, const char *dst, bool append) {
   return status;
 }
 
-void nntrainer_test() {
-  ml_train_model_h model = NULL;
+void data_train_model() {
+  ml_train_model_h model;
+  char model_conf_path[PATH_MAX];
+  char label_path[PATH_MAX];
+  int status = ML_ERROR_NONE;
+  FILE *file;
+
+  data_get_resource_path("model.ini", model_conf_path, false);
+  data_get_data_path("label.dat", label_path);
+
+  file = fopen(label_path, "w");
+  if (file == NULL) {
+    _E("Error opening file");
+    return;
+  }
+
+  if (fputs("sad\nsmile\n\n", file) < 0) {
+    _E("error writing");
+    fclose(file);
+    return;
+  }
+
+  if (fclose(file) < 0) {
+    _E("Error closing file");
+    return;
+  }
+
+  _D("model conf path: %s", model_conf_path);
+
+  status = ml_train_model_construct_with_conf(model_conf_path, &model);
+  if (status != ML_ERROR_NONE) {
+    _E("constructing trainer model failed %d", status);
+    goto CLEAN_UP;
+  }
+
+  status = ml_train_model_compile(model, NULL);
+  if (status != ML_ERROR_NONE) {
+    _E("compile model failed %d", status);
+    goto CLEAN_UP;
+  }
+
+  status = ml_train_model_run(model, NULL);
+  if (status != ML_ERROR_NONE) {
+    _E("run model failed %d", status);
+    goto CLEAN_UP;
+  }
+
+  status = ml_train_model_destroy(model);
+  if (status != ML_ERROR_NONE) {
+    _E("Destryoing model failed %d", status);
+  }
 
-  ml_train_model_construct(model);
+CLEAN_UP:
   ml_train_model_destroy(model);
```

Comment:
log return valus of `ml_train_model_destroy` if this also fails?

Reply:
It is done  Thanks